prefeRx/src/test/java/com/vinted/preferx/EnumPreferenceImplTest.kt

```diff
@@ -0,0 +1,45 @@
+package com.vinted.preferx
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.test.core.app.ApplicationProvider
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class EnumPreferenceImplTest {
+
+
+    private val appContext = ApplicationProvider.getApplicationContext<Context>()
+    private val sharedPref = appContext.getSharedPreferences("TestPrefs", Context.MODE_PRIVATE)
+    private val sharedPrefEditor: SharedPreferences.Editor = sharedPref.edit()
+    private val fixture = EnumPreferenceImpl(sharedPref, "key", TestEnum.BAR)
```

Comment:
Another way would be going for a stateless approach for more flexibility, and also 'hide' implementation details:
```
    private fun createTestEnumPreference(default: TestEnum = TestEnum.BAR): EnumPreference<TestEnum> {
        val sharedPreferences = ApplicationProvider
            .getApplicationContext<Context>()
            .getSharedPreferences("TestPrefs", Context.MODE_PRIVATE)

        return EnumPreferenceImpl(sharedPreferences, "key", default)
    }
    
    // And then in test
    val fixture = createTestEnumPreference()
    ...
```

One benefit is that tests will be able to evolve independently of each other, without every test having to use the same configuration.